packages/py-api/py_api/controllers/hackathon/participants_controller.py

```diff
@@ -240,64 +241,70 @@ def add_participant(cls, participant: NewParticipant, jwt_token: str | None = No
 
             # Find the teams which can accept a new participant
             for team in random_teams:
-
                 if (len(team.team_members) < 6):
-                    avaliable_team = team
-                    try:
-                        # Creates new participant
-                        new_participant = ParticipantsFunctionality.create_participant(
-                            participant,
-                        )
-
-                        new_participant_object_id = str(
-                            new_participant.inserted_id,
-                        )
-                        # Adds the participant to the team
-                        avaliable_team = TeamFunctionality.add_participant_to_team_object(
-                            avaliable_team.team_name, new_participant_object_id,
-                        )
-                        cls.update_participant(
-                            new_participant_object_id, UpdateParticipant(
-                                team_name=avaliable_team.team_name,
-                            ),
-                        )
-                        TeamFunctionality.update_team_query_using_dump(
-                            avaliable_team.model_dump(),
-                        )
-                        return JSONResponse(content=avaliable_team.model_dump(), status_code=200)
-
-                    except (Exception) as e:
-                        return JSONResponse(content={"message": str(e)}, status_code=500)
-
-            # If all the random teams are full, it checks if there is space for creating a new one
+                    return cls.add_participant_to_existing_random_team(team, participant)
+
+            # If all the random teams are full, check if there is space for creating a new one
             if (TeamFunctionality.get_count_of_teams() < 15):
```

Comment:
Move that check and the block of code before the for loop on line 243

Reply:
Which check?